analysis/repeat_events/process_repeat_events.R

```diff
@@ -93,7 +82,10 @@ for (cohort in c("prevax", "unvax", "vax")) {
       values_drop_na = TRUE
     ) %>%
     # tidy up the outcome column so it only shows the event type
-    mutate(across(outcome, ~str_remove_all(.x, "out_date_|_\\d+")))
+    mutate(across(outcome, ~str_remove_all(.x, "out_date_|_\\d+"))) %>%
+    # drop outcome events that don't happen between index_date and end_date_outcome (inclusive)
+    filter(between(out_date, index_date, end_date_outcome)) %>%
```

Comment:
It's more efficient to replace the removed lines above with this `filter(between())` statement